packages-new/components/checkbox/src/components/osds-checkbox/osds-checkbox.spec.ts

```diff
@@ -0,0 +1,285 @@
+// jest.mock('@ovhcloud/ods-core/src/components/checkbox/ods-checkbox-controller'); // keep jest.mock before any import
+
+import type { OdsCheckboxAttribute } from './interfaces/attributes';
+import type { SpecPage } from '@stencil/core/testing';
+import { newSpecPage } from '@stencil/core/testing';
+import { OsdsCheckbox } from './osds-checkbox';
+import {
+  odsComponentAttributes2StringAttributes,
+  odsGetSimulatedPromise,
+  OdsMockNativeMethod,
+  odsStringAttributes2Str,
+  odsUnitTestAttribute,
+  OdsUnitTestAttributeType,
+} from '@ovhcloud/ods-common-testing';
+import { OdsCheckboxController } from './core/ods-checkbox-controller';
+import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
+
+describe('spec:osds-checkbox', () => {
+  let page: SpecPage;
+  let htmlCheckbox: (OsdsCheckbox & HTMLElement) | null | undefined;
+  let mainSlot: HTMLElement | null | undefined;
+  let instance: OsdsCheckbox;
+  let shadowRoot: ShadowRoot | null | undefined;
+  let controller: OdsCheckboxController;
+  /** base html template (avoid boilerplate) */
+  const baseHtml = (slots: { unnamed?: string }) =>
+    `
+    ${slots.unnamed || slots.unnamed === '' ? slots.unnamed : `<input type="checkbox">`}
+    `;
+
+  async function setup({ attributes = {}, html = `` }: { attributes?: Partial<OdsCheckboxAttribute>, html?: string }) {
+    const baseAttribute = { ariaLabel: '', checked: false, disabled: false, hasFocus: false, updating: false, value: '' };
+    const stringAttributes = odsComponentAttributes2StringAttributes<OdsCheckboxAttribute>({ ...baseAttribute, ...attributes }, DEFAULT_ATTRIBUTE);
+
+    // mock setCustomValidity method that does not exist when stencil mock HTMLInputElement
+    OdsMockNativeMethod(HTMLInputElement.prototype, "focus", jest.fn());
+
+    page = await newSpecPage({
+      components: [OsdsCheckbox],
+      html: `<osds-checkbox ${odsStringAttributes2Str(stringAttributes)}>${html}</osds-checkbox>`,
+    });
+
+    instance = page.rootInstance;
+
+    shadowRoot = page.root?.shadowRoot;
+    htmlCheckbox = shadowRoot?.querySelector('input[type=checkbox]');
+
+    // note: assigned slot not yet available in HtmlMockedElement of stencil : https://github.com/ionic-team/stencil/issues/2830
+    mainSlot = shadowRoot?.querySelector('slot:not([name])');
+
+    // controller = (OdsCheckboxController as unknown as jest.SpyInstance<OdsCheckboxController, unknown[]>).mock.instances[ 0 ];
+  }
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  fit('should render', async () => {
```

Comment:
remove `f`

Reply:
Francois create a card for : https://jira.ovhcloud.tools/browse/ODS-839

Reply:
Francois create a card for this ODS-839